ddtrace/llmobs/_integrations/langchain.py

```diff
@@ -260,6 +263,57 @@ def _llmobs_set_meta_tags_from_embedding(
             except (TypeError, IndexError):
                 log.warning("Failed to write output vectors", output_embedding)
 
+    def _llmobs_set_meta_tags_from_similarity_search(
+        self,
+        span: Span,
+        input_query: str,
+        output_documents: Union[List[Any], None],
+        error: bool = False,
+        is_workflow: bool = False,
+    ) -> None:
+        span.set_tag_str(SPAN_KIND, "workflow" if is_workflow else "retrieval")
+        span.set_tag_str(MODEL_NAME, span.get_tag(MODEL) or "")
+        span.set_tag_str(MODEL_PROVIDER, span.get_tag(PROVIDER) or "")
+
+        if input_query is not None:
+            try:
+                formatted_inputs = self.format_io(input_query)
+                if isinstance(formatted_inputs, str):
+                    span.set_tag_str(INPUT_VALUE, formatted_inputs)
+                else:
+                    span.set_tag_str(INPUT_VALUE, json.dumps(formatted_inputs))
+            except TypeError:
+                log.warning("Failed to serialize similarity search input to JSON")
+        if error:
+            span.set_tag_str(OUTPUT_VALUE, "")
+        elif isinstance(output_documents, list):
+            if is_workflow:
+                span.set_tag_str(OUTPUT_VALUE, "[{} document(s) retrieved]".format(len(output_documents)))
+            else:
+                documents = []
+                for d in output_documents:
+                    doc = Document(text=d.page_content)
+                    try:
+                        doc["id"] = d.id
+                    except AttributeError:
+                        log.warning("Failed to extract document id from similarity search output")
+                        doc["id"] = ""
+                    try:
+                        doc["name"] = d.metadata["name"]
+                    except KeyError:
+                        log.warning("Failed to extract document name metadata from similarity search output")
+                        try:
+                            doc["name"] = d.id
+                        except AttributeError:
```

Comment:
Are these fields not guaranteed to exist on Langchain Document objects? I think we're more concerned whether or not the values are `None`. Instead of try/catch attribute/key erros, can we do something like this?
```
doc["id"] = d.id or ""
```
For the `name` field we can keep the key error catch.